sympy/polys/matrices/ddm.py

```diff
@@ -92,6 +92,12 @@ def __init__(self, rowslist, shape, domain):
         if not (len(self) == m and all(len(row) == n for row in self)):
             raise DDMBadInputError("Inconsistent row-list/shape")
 
+    def extract_slice(self, slice1, slice2):
+        r = slice1.stop-slice1.start
+        c = slice2.stop-slice2.start
+        ddm = [row[slice2] for row in self[slice1]]
+        return self.new(ddm, (r, c), self.domain)
```

Comment:
There needs to be tests for this. It doesn't look like it would correctly handle a step that isn't one e.g. `M[::-1, :]`

Reply:
Sure, the latest commit fixes this issue.
Thanks.